src/scheduler.cc

```diff
@@ -255,19 +254,19 @@ Status SchedulerService::KillWorkers(ServerContext* context, const KillWorkersRe
   size_t busy_workers = 0;
   std::vector<WorkerHandle*> idle_workers;
   RAY_LOG(RAY_INFO, "Attempting to kill workers.");
-  for (size_t i = 0; i < workers->size(); ++i) {
-    WorkerHandle* worker = &(*workers)[i];
-    if (worker->worker_stub) {
-      if (worker->current_task == NO_OPERATION) {
-        idle_workers.push_back(worker);
-        RAY_CHECK(std::find(avail_workers->begin(), avail_workers->end(), i) != avail_workers->end(), "Worker with workerid " << i << " is idle, but is not in avail_workers_");
-        RAY_LOG(RAY_INFO, "Worker with workerid " << i << " is idle.");
-      } else if (worker->current_task == ROOT_OPERATION) {
+  for (auto i = workers->begin(); i != workers->end(); ++i) {
+    auto worker_id = static_cast<WorkerId>(i - workers->begin());
```

Comment:
It's just like pointer subtraction, it gives you the distance between the iterators; it's meant for this purpose.